src/term/mod.rs

```diff
@@ -233,40 +232,34 @@ impl<'a> RenderableCellsIter<'a> {
         }
     }
 
-    /// Populates list of cursor cells with the original cell
+    /*/// Populates list of cursor cells with the original cell
     fn populate_no_cursor(&mut self) {
         self.cursor_cells.push_back(Indexed {
             line: self.cursor.line,
             column: self.cursor.col,
             inner: self.grid[self.cursor],
         });
-    }
+    }*/
 
     fn initialize(mut self, cursor_style: CursorStyle) -> Self {
-        if self.cursor_is_visible() {
+        if let Some(cursor) = self.cursor {
             match cursor_style {
                 CursorStyle::Block => {
-                    self.populate_block_cursor();
+                    self.populate_block_cursor(&cursor);
                 },
                 CursorStyle::Beam => {
-                    self.populate_beam_cursor();
+                    self.populate_beam_cursor(&cursor);
                 },
                 CursorStyle::Underline => {
-                    self.populate_underline_cursor();
+                    self.populate_underline_cursor(&cursor);
                 }
             }
         } else {
-            self.populate_no_cursor();
+            //self.populate_no_cursor();
```

Comment:
You've removed the fallback "no_cursor" here. Is this planned to be upstreamed or should this functionality be added again?

Reply:
To be honest, I'm not exactly sure what I was thinking. It was introduced in [this commit]( https://github.com/neon64/alacritty/commit/4b224948333c55a4bda1c40ba2ad621b6b7be724). I think perhaps I was under the impression that if the cursor is off the screen or the `SHOW_CURSOR` mode isn't set, then we don't need to show it => no work needs to be done. Perhaps @jwilm could explain why it was necessary that a "no cursor" needed to be rendered in the first place?